pynicotine/gtkgui/search.py

```diff
@@ -1503,20 +1505,26 @@ def on_refilter(self, *_args):
             except re.error:
                 filter_out = None
 
+        # Split at | pipes ampersands & space(s) but don't split operators spaced before digit
+        seperator_pattern = re.compile(r"(?:[|&\s])+(?<![<>!=]\s)")
```

Comment:
You can compile these patterns in a class constant instead of doing it here every time.

Reply:
Thank you, I didn't know you could do that.